core/sys/orca/orca.odin

```diff
@@ -0,0 +1,2183 @@
+	reserve: u64,
+}
+
+@(default_calling_convention="c", link_prefix="oc_")
+foreign {
+	// Initialize a memory arena.
+	arena_init :: proc(arena: ^arena) ---
+	// Initialize a memory arena with additional options.
+	arena_init_with_options :: proc(arena: ^arena, options: ^arena_options) ---
+	// Release all resources allocated to a memory arena.
+	arena_cleanup :: proc(arena: ^arena) ---
+	// Allocate a block of memory from an arena.
+	arena_push :: proc(arena: ^arena, size: u64) -> rawptr ---
+	// Allocate an aligned block of memory from an arena.
+	arena_push_aligned :: proc(arena: ^arena, size: u64, alignment: u32) -> rawptr ---
+	// Reset an arena. All memory that was previously allocated from this arena is released to the arena, and can be reallocated by later calls to `oc_arena_push` and similar functions. No memory is actually released _to the system_.
+	arena_clear :: proc(arena: ^arena) ---
+	// Begin a memory scope. This creates an `oc_arena_scope` object that stores the current offset of the arena. The arena can later be reset to that offset by calling `oc_arena_scope_end`, releasing all memory that was allocated within the scope to the arena.
+	arena_scope_begin :: proc(arena: ^arena) -> arena_scope ---
+	// End a memory scope. This resets an arena to the offset it had when the scope was created. All memory allocated within the scope is released back to the arena.
+	arena_scope_end :: proc(scope: arena_scope) ---
+	/*
+	Begin a scratch scope. This creates a memory scope on a per-thread, global "scratch" arena. This allows easily creating temporary memory for scratch computations or intermediate results, in a stack-like fashion.
+	
+	If you must return results in an arena passed by the caller, and you also use a scratch arena to do intermediate computations, beware that the results arena could itself be a scatch arena. In this case, you have to be careful not to intermingle your scratch computations with the final result, or clear your result entirely. You can either:
+	
+	- Allocate memory for the result upfront and call `oc_scratch_begin` afterwards, if possible.
+	- Use `oc_scratch_begin_next()` and pass it the result arena, to get a scratch arena that does not conflict with it.
+	*/
+	scratch_begin :: proc() -> arena_scope ---
+	// Begin a scratch scope that does not conflict with a given arena. See `oc_scratch_begin()` for more details about when to use this function.
+	scratch_begin_next :: proc(used: ^arena) -> arena_scope ---
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// String slices and string lists.
+////////////////////////////////////////////////////////////////////////////////
+
+// A type representing a string of bytes.
+str8 :: string
+
+// A type representing an element of a string list.
+str8_elt :: struct {
+	// The string element is linked into its parent string list through this field.
+	listElt: list_elt,
+	// The string for this element.
+	_string: str8,
```

Comment:
Any reason this is `_string` and not just `string`?

Reply:
mostly temporary, thought it was not allowed but that was just syntax highlighting